### PR TITLE
VB-2441 Update session template with incentive types

### PR DESCRIPTION
**Pull request by AledWEvans**
VB-2441 Update session template, add form - incentive types

### CODE REVIEW

File: server/routes/prisons/sessionTemplates/addSessionTemplateController.ts [hunk 1]
```diff
@@ -45,6 +50,13 @@ export default class AddSessionTemplateController {
       const validToDateMonth = validToDateSplit[1] || undefined
       const validToDateDay = validToDateSplit[2] || undefined
 
+      let incentiveGroupReferences: string[] = []
+      const incentiveGroups = sessionTemplate.prisonerIncentiveLevelGroups
+      if (incentiveGroups !== undefined && incentiveGroups.length > 0) {
+        incentiveGroupReferences = incentiveGroups.map(incentiveGroup => incentiveGroup.reference)
+      }
+
+      logger.debug(` group refs : ${JSON.stringify(incentiveGroupReferences)} ${JSON.stringify(incentiveGroups)}`)
```
Comment (tpmcgowan) [inline server/routes/prisons/sessionTemplates/addSessionTemplateController.ts:59]: Some stray debug logging...

File: integration_tests/pages/page.ts [hunk 2]
```diff
@@ -26,4 +26,12 @@ export default abstract class Page {
   signOut = (): PageElement => cy.get('[data-qa=signOut]')
 
   manageDetails = (): PageElement => cy.get('[data-qa=manageDetails]')
+
+  getByClass = (clazz: string): PageElement => cy.get(`.${clazz}]`)
```
Comment (tpmcgowan) [inline integration_tests/pages/page.ts:30]:
```suggestion
  getByClass = (class: string): PageElement => cy.get(`.${class}]`)
```
Though the original version still works 😀

File: integration_tests/pages/prisons/addSessionTemplatePage.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+  getDayOfWeekInput = () => this.getById('dayOfWeek')
+
+  getStartTimeInput = () => this.getById('startTime')
+
+  getEndTimeInput = () => this.getById('endTime')
+
+  getWeeklyFrequencyInput = () => this.getById('weeklyFrequency')
+
+  enterValidFromDate = (date: Date) => {
```
Comment (tpmcgowan) [inline integration_tests/pages/prisons/addSessionTemplatePage.ts:22]: Nice!

**Comment by tpmcgowan** [comment]
Looks good 👍 - just a few minor comments. Haven't approved as it looks like there are conflicts to fix....

File: server/routes/prisons/sessionTemplates/addSessionTemplateController.ts (hunk 1 above)
Comment (AledWEvans) [reply server/routes/prisons/sessionTemplates/addSessionTemplateController.ts:59]: will fix

File: integration_tests/pages/page.ts (hunk 2 above)
Comment (AledWEvans) [reply integration_tests/pages/page.ts:30]: thats the cool why of saying it ;-)

**Review by tpmcgowan — APPROVED** [review]
LGTM!

**Pull request merged**